client/blocks/video-editor/video-editor-upload-button.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import ReactDom from 'react-dom';
+import React, { Component, PropTypes } from 'react';
+import { noop } from 'lodash';
+import classNames from 'classnames';
+
+class VideoEditorUploadButton extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		isPosterUpdating: PropTypes.bool,
+		onClick: PropTypes.func,
+		onUploadImage: PropTypes.func,
+	};
+
+	static defaultProps = {
+		isPosterUpdating: false,
+		onClick: noop,
+		onUploadImage: noop,
+	};
+
+	setFormInstance = ref => this.form = ref;
+
+	handleChange = ( event ) => {
+		let files;
+
+		if ( event.target.files && event.target.files.length > 0 ) {
+			files = event.target.files[ 0 ];
+		}
+
+		this.props.onUploadImage( files );
+		ReactDom.findDOMNode( this.form ).reset();
```

Comment:
Can we think of a way to avoid this? Perhaps we can control the `value`prop of the input, or add a key prop to force a new instance to render:
```
<input
    key={ this.state.key }
```